### PR TITLE
Drop Support for Python 3.7

### PR DESCRIPTION
**Pull request by jules-ch**
- Update CHANGES
- Update README
- Add black badge in README
- Update Docs
- Move pull request template to github folder
- Update CI with minimal dependencies

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,21 +7,18 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7, 3.8, 3.9]
-        numpy: [null, "numpy>=1.17,<2.0.0"]
-        uncertainties: [null, "uncertainties==3.1.4", "uncertainties>=3.1.4,<4.0.0"]
+        python-version: [3.8, 3.9, "3.10"]
+        numpy: [null, "numpy>=1.19.5,<2.0.0"]
```
Comment (keewis) [inline .github/workflows/ci.yml:11]: is there a reason for requiring `numpy>=1.19.5` instead of `numpy>=1.19`? Or is this is from the min-versions job?

**Review by keewis — COMMENTED** [review]
I have a single comment, otherwise this looks good to me

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jules-ch) [reply .github/workflows/ci.yml:11]: No just want to be sure to use a numpy patched version.

**Comment by jules-ch** [comment]
@hgrecco you're ok with this ?

**Comment by hgrecco** [comment]
Sounds good to me.!

**Comment by jules-ch** [comment]
Ok It is merged! :+1:

**Pull request merged**